td-logger/src/logger.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+
+//! A standalone logger to directly log messages, independent of the log crate.
+
+use core::fmt::{self, Write};
+use lazy_static::lazy_static;
+use spin::Mutex;
+
+use crate::{dbg_port_write, dbg_write_string};
+
+pub const LOG_LEVEL_VERBOSE: usize = 1000;
+pub const LOG_LEVEL_INFO: usize = 100;
+pub const LOG_LEVEL_WARN: usize = 10;
+pub const LOG_LEVEL_ERROR: usize = 1;
+pub const LOG_LEVEL_NONE: usize = 0;
+
+pub const LOG_MASK_COMMON: u64 = 0x1;
+// Core - Boot Service (BIT1 ~ BIT15)
+pub const LOG_MASK_PROTOCOL: u64 = 0x2;
+pub const LOG_MASK_MEMORY: u64 = 0x4;
+pub const LOG_MASK_EVENT: u64 = 0x8;
+pub const LOG_MASK_IMAGE: u64 = 0x10;
+// Core - Runtime Service (BIT16 ~ BIT 23)
+pub const LOG_MASK_VARIABLE: u64 = 0x10000;
+// Core - Console (BIT24 ~ BIT 31)
+pub const LOG_MASK_CONOUT: u64 = 0x1000000;
+pub const LOG_MASK_CONIN: u64 = 0x2000000;
+// Protocol - (BIT32 ~ BIT63)
+pub const LOG_MASK_BLOCK_IO: u64 = 0x100000000;
+pub const LOG_MASK_FILE_SYSTEM: u64 = 0x200000000;
```

Comment:
Above (from LOG_MASK_PROTOCOL to LOG_MASK_FILE_SYSTEM) seems UEFI concept.

I am not sure they are absolutely needed.

To simplify the design, how about we can remove them at this moment ?
If there is need, we can add based upon real use case.

Reply:
Sounds reasonable.